src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -93,14 +93,25 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
 
         ov::Tensor inputs_embeds = m_inputs_embedder->get_inputs_embeds(prompt, rgbs);
 
+        auto to_remove_from_hist = m_inputs_embedder->get_amount_to_remove_from_hist();
+        if (to_remove_from_hist > 0) {
+            ov::genai::utils::trim_kv_cache(m_language, to_remove_from_hist);
+        }
+
         Sampler sampler = Sampler(m_tokenizer);
 
         std::vector<SequenceGroup::Ptr> requests;
         size_t request_id = 0;
         size_t block_size = 1; // not used
         bool enable_prefix_caching = false;
+
+        auto tokenized_chat_history = m_inputs_embedder->get_tokenized_chat_history();
         size_t history_size = m_language.get_tensor("attention_mask").get_shape().at(1);
         size_t inputs_embeds_size = inputs_embeds.get_shape().at(1);
+
+        if (to_remove_from_hist > 0)
```

Comment:
I suppose it's always >= 0 as `get_amount_to_remove_from_hist` returns `size_t` , so it means we can subtract unconditionally ?